service-vs-library.md

```diff
@@ -0,0 +1,82 @@
+## Title
+Service vs. library: It's inner source, not inner deployment
+
+## Patlet
+Teams in a DevOps environment may be reluctant to work across team boundaries on
+interdependent code bases due to ambiguity over who will be responsible for
+responding to service downtime. The solution is to realize that often it's
+possible to deploy the same service in independent environments with separate
+escalation chains in the event of service downtime.
+
+## Problem
+
+When teams are working in a DevOps environment developers are responsible for a
+feature end-to-end: From the customer down to deployment, maintenance and
+support. This poses a challenge when working across team boundaries: Escalation
+chains may not be the same for errors happening in either team. Coupling
+source code and deployment leaves the teams with the question of who is
+responsible for on-call support in the event of errros. As a result teams are
+reluctant to join forces even if there is significant overlap in requirements.
+
+## Context
+
+Teams are working in a micro-services environment.
+
+They are organised in full functional DevOps teams: Each team is responsible for
+their contributions end-to-end, including maintenance, on-call and customer
+support.
+
+A team is tasked with providing a service to their downstream customers that is
+fairly similar to an existing service built by another team.
+
+## Forces
+
+Organisational escalation paths may be different for each of the teams.
+
+Members of each team may be unwilling to answer on-call support for errors that
+do not affect their own downstream customers.
+
+Severity levels for the same types of errors may be different across team
+boundaries due to different SLA definitions per team/customer relationship.
+
+## Solutions
+
+Decouple responsibility for source code from deployment: Both teams work to
+identify exactly where there is overlap and synergies.
+
+Only shared source code is kept as part of the InnerSource project with shared
+responsiblity.
+
+Decouple configuration and deployment pipelines from actual business logic.
+Establish a second deployment of the service for the second team.
+
+Treat the common base as a library that is used by both teams with shared code
```

Comment:
The **Patlet** talks about `deploying the same service in independent environments` yet the solution talks about factoring out shared service code into a library, which becomes the focus of collaboration (not the service).  I feel like these two sections of the document are incongruent with each other.  Are the teams collaborating on the source code for a library or a service?